client/src/components/relatedProducts/Carousel.jsx

```diff
@@ -9,31 +9,39 @@ import './Carousel.css'
 
 export default function Carousel(props) {
   const [relatedProducts, setRelatedProducts] = useState([])
-  const [productDetails, setProductDetails] = useState([])
+  const [currentProduct, setCurrentProduct] = useState([])
   const [currentIndex, setCurrentIndex] = useState(0)
 
 
-
   const currentRelated = relatedProducts.length
   const { productId } = useParams()
 
   useEffect(() => {
-    getRelatedProducts()
+    let mounted = true;
+    if (mounted) {
+      getRelatedProducts()
+      getCurrentProduct()
+    }
+    return function cleanup() {
+      mounted = false;
+    }
   }, [])
 
   const getRelatedProducts = async () => {
     try {
       const { data } = await axios.get(`/product/${productId}/related`)
       setRelatedProducts(data)
+      console.log('related', data)
     } catch (error) {
       console.log(error.message)
     }
   }
 
-  const getProductDetails = async () => {
+  const getCurrentProduct = async () => {
     try {
-      const { data } = await axios.get(`/api/products/${productId}`)
-      setProductDetails(data)
+      const { data } = await axios.get(`/reviews/meta?product_id=${productId}`)
+      setCurrentProduct(data)
+      console.log('current', data)
```

Comment:
I would recommend removing console logs before merging to main. especially when it is logging data available in state and thus the react dev tools.